Mathlib/Algebra/Group/Fin/Basic.lean

```diff
@@ -107,9 +107,59 @@ lemma lt_sub_one_iff {k : Fin (n + 2)} : k < k - 1 ↔ k = 0 := by
 lemma sub_one_lt_iff {k : Fin (n + 1)} : k - 1 < k ↔ 0 < k :=
   not_iff_not.1 <| by simp only [lt_def, not_lt, val_fin_le, le_sub_one_iff, le_zero_iff]
 
+lemma lt_sub_iff {n : ℕ} {a b : Fin n} : a < a - b ↔ a < b := by
+  cases' n with n
+  · exact a.elim0
+  constructor
+  · contrapose!
+    intro h
+    simp only [Fin.not_lt] at h ⊢
+    obtain ⟨l, hl⟩ := Nat.exists_eq_add_of_le h
+    rw [le_iff_val_le_val, sub_def, hl]
+    simp only
+    rw [← Nat.add_assoc, Nat.sub_add_cancel b.is_lt.le, Nat.add_mod_left, Nat.mod_eq_of_lt]
+    · simp
+    · refine a.is_lt.trans_le' ?_
+      simp [hl]
+  · intro h
+    rw [lt_iff_val_lt_val, sub_def]
+    simp only
+    obtain ⟨k, hk⟩ := Nat.exists_eq_add_of_lt b.is_lt
+    have : n + 1 - b = k + 1 := by
+      simp_rw [hk] -- simp_rw because, otherwise, rw tries to rewrite inside `b : Fin (n + 1)`
+      rw [Nat.add_assoc, Nat.add_sub_cancel_left]
+    rw [this, Nat.mod_eq_of_lt]
+    · simp
+    · refine hk.ge.trans_lt' ?_
+      simp [Nat.add_assoc, h, Nat.add_comm, Nat.add_left_comm]
```

Comment:
```suggestion
    cases' n with n
  · exact a.elim0
  constructor
  · contrapose!
    intro h
    obtain ⟨l, hl⟩ := Nat.exists_eq_add_of_le (Fin.not_lt.mp h)
    simpa only [Fin.not_lt, le_iff_val_le_val, sub_def, hl, ← Nat.add_assoc, Nat.add_mod_left,
      Nat.mod_eq_of_lt, Nat.sub_add_cancel b.is_lt.le] using
        (le_trans (mod_le _ _) (le_add_left _ _))
  · intro h
    rw [lt_iff_val_lt_val, sub_def]
    simp only
    obtain ⟨k, hk⟩ := Nat.exists_eq_add_of_lt b.is_lt
    have : n + 1 - b = k + 1 := by
      simp_rw [hk, Nat.add_assoc, Nat.add_sub_cancel_left] 
      -- simp_rw because, otherwise, rw tries to rewrite inside `b : Fin (n + 1)`
    rw [this, Nat.mod_eq_of_lt (hk.ge.trans_lt' ?_), Nat.lt_add_left_iff_pos] <;>
    omega
```